src/core/client/client.internal.ts

```diff
@@ -28,11 +40,110 @@ export class InternalClient {
     private readonly logger?: ILogger
   ) {}
 
+  /**
+   * Resolves profile AST file.
+   * File property:
+   *  - loads directly passed file
+   *  - can point only to .supr.ast.json file
+   *  - throws if file not found or not valid ProfileDocumentNode
+   * Version property:
+   *  - looks for [profileId]@[version].supr.ast.json file in superface/grid
+   *  - if not found it tries to fetch profile AST from Registry
+   * @returns ProfileDocumentNode
+   */
+  public async resolveProfileAst(
```

Comment:
question: does this need to be a method on the client?

Reply:
Probably not. I wasn't sure where to move it.

Reply:
@lukas-valenta should I move it? 😅

Reply:
yeah, move it to a function near Profile